projects/plugins/jetpack/modules/shortcodes/gravatar.php

```diff
@@ -93,16 +98,60 @@ function jetpack_gravatar_profile_shortcode( $atts ) {
 		return false;
 	}
 
+	$hashed_email = hash( 'sha256', strtolower( trim( $user->user_email ) ) );
+	$cache_key    = 'grofile-' . $hashed_email;
+	$profile      = get_transient( $cache_key );
+
+	if ( empty( $profile ) ) {
+		$profile_url = sprintf(
+			'https://secure.gravatar.com/%s.json',
+			$hashed_email
+		);
+
+		$expire        = 300;
+		$response      = wp_remote_get(
+			esc_url_raw( $profile_url ),
+			array( 'User-Agent' => 'WordPress.com Gravatar Profile Widget' )
+		);
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( $response_code === 200 ) {
+			$profile = wp_remote_retrieve_body( $response );
+			$profile = json_decode( $profile, true );
+
+			if ( is_array( $profile ) && ! empty( $profile['entry'] ) && is_array( $profile['entry'] ) ) {
+				// Cache for 15 minutes.
+				$expire  = 900;
+				$profile = $profile['entry'][0];
+			} else {
+				// Something strange happened.  Cache for 5 minutes.
+				$profile = array();
+			}
+		} else {
+			// Cache for 15 minutes.
+			$expire  = 900;
+			$profile = array();
+		}
+
+		set_transient( $cache_key, $profile, $expire );
+	}
+
+	if ( empty( $profile ) ) {
+		return false;
+	}
+
 	// Render the shortcode.
-	$gravatar_url = 'https://gravatar.com/' . $user->user_login;
+	$gravatar_url     = 'https://gravatar.com/' . $hashed_email;
+	$user_location    = ! empty( $profile['currentLocation'] ) ? $profile['currentLocation'] : '';
+	$display_name     = ! empty( $profile['displayName'] ) ? $profile['displayName'] : '';
+	$user_description = ! empty( $profile['aboutMe'] ) ? $profile['aboutMe'] : '';
 
 	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
-		$avatar_url    = wpcom_get_avatar_url( $user->ID, 96 );
-		$avatar_url    = $avatar_url[0];
-		$user_location = get_user_attribute( $user->ID, 'location' );
+		$gravatar_url = 'https://gravatar.com/' . $user->user_login;
```

Comment:
Is there a point in having 2 different scenarios for `$gravatar_url`? Given that the <gravatar.com/hash> redirects, maybe we could always use the <gravatar.com/hash> format, to simplify things?

Reply:
I left it to avoid a redirect if possible, but I can update it if feels cleaner.

Reply:
That makes sense. It doesn't really make things that much more complex, so I'm not insisting on this. 👍